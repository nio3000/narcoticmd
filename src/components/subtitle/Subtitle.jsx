import React from 'react';
import PropTypes from 'prop-types';
import './Subtitle.css';

const Subtitle = ({value}) =>
    Boolean(value) && <h2 className="subtitle">{ value }</h2>;

Subtitle.propTypes = {
  value: PropTypes.node
};
export default Subtitle;
