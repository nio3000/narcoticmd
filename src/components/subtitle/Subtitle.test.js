import React from 'react';
import renderer from 'react-test-renderer';

import Subtitle from './Subtitle';

describe('Subtitle', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Subtitle value="Editor for markdown-addicted writers"/>
    );
    expect( component.toJSON() ).toMatchSnapshot();
  });
  it('doesn\'t render if its value is empty', () => {
    const component = renderer.create(
      <Subtitle value=""/>
    );
    expect( component.toJSON() ).toMatchSnapshot();
  })
});