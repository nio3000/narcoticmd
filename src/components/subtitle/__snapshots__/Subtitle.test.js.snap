// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle doesn't render if its value is empty 1`] = `null`;

exports[`Subtitle renders correctly 1`] = `
<h2
  className="subtitle"
>
  Editor for markdown-addicted writers
</h2>
`;
